File: Models/Payment.cs
namespace POS.Models;

enum PaymentType
{
    CASH,
    CREDIT_CARD
}
public class Payment
{
    public int Id { get; set; }
    public float Amount { get; set; }
    public PaymentType PaymentMethod { get; set; }
    public DateTime PaidAt { get; set; }
}

File: payment-service/src/main/java/solutions/skydev/pos/payment_service/model/entity/Payment.java
package solutions.skydev.pos.payment_service.model.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Setter
@Getter
@NoArgsConstructor
@Entity
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private BigDecimal amount;
    // TODO: make enum
    private String paymentMethod;
    private String notes;
    private Long billingRequestId;
    private String status = "paid"; // default status is 'paid'
}
